include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -28,39 +28,49 @@ struct NonUniformBSplinesBase
 
 } // namespace detail
 
-/// NonUniformPointSampling specialization of BSplines
+/**
+ * The type of a non-uniform B-splines 1D basis.
+ *
+ * @tparam Tag The tag identifying the continuous dimension on which the support of the B-spline functions are defined.
+ * @tparam D The degree of the B-splines.
+ */
 template <class Tag, std::size_t D>
 class NonUniformBSplines : detail::NonUniformBSplinesBase
 {
     static_assert(D > 0, "Parameter `D` must be positive");
 
 public:
+    /// @brief The tag identifying the continuous dimension on which the support of the B-splines are defined.
     using tag_type = Tag;
 
-
+    /// @brief The discrete dimension identifying B-splines.
     using discrete_dimension_type = NonUniformBSplines;
 
-public:
+    /// @brief The degree of B-splines.
     static constexpr std::size_t degree() noexcept
     {
         return D;
     }
 
+    /// @brief Indicates if the B-splines are periodic or not.
     static constexpr bool is_periodic() noexcept
     {
         return Tag::PERIODIC;
     }
 
+    /// @brief Indicates if the B-splines are radial or not (should be deprecated soon because this concept is not in the scope of DDC).
     static constexpr bool is_radial() noexcept
```

Comment:
I agree, i don't think it used anywhere.
```suggestion
     [[deprecated]] static constexpr bool is_radial() noexcept
```